### PR TITLE
ISPN-6997PessimisticTxPartitionAndMergeDuringRuntimeTest

### PR DESCRIPTION
**Pull request by danberindei**
.testOriginatorIsolatedPartition random failures

https://issues.jboss.org/browse/ISPN-6997

On one node, transactionTable.cleanupLeaverTransactions was being
called only before the split and after the merge, never with the list
of members during the split.

### CODE REVIEW

**Comment by danberindei** [comment]
I've looked some more into this issue and it's more complex than I thought: a remote tx command can create a new remote transaction even when the tx originator is in another partition. If the partitions merge immediately afterwards, neither the originator nor the remote node's `TransactionTable` will clean up the transaction.

File: core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java [hunk 1]
```diff
@@ -277,37 +277,41 @@ public int getMinTopologyId() {
    }
 
    public void cleanupLeaverTransactions(List<Address> members) {
-      // Can happen if the cache is non-transactional
-      if (remoteTransactions == null)
-         return;
+      try {
+         // Can happen if the cache is non-transactional
+         if (remoteTransactions == null)
```
Comment (slaskawi) [inline core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java]: Why did you move it inside the `try` block?

Comment (danberindei) [inline core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java]: I added the try block to make sure we don't throw any exception here that would interrupt the topology update in `StateConsumerImpl`.

Comment (slaskawi) [inline core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java]: We won't throw anything with the `return` instruction ;)

Comment (danberindei) [inline core/src/main/java/org/infinispan/transaction/impl/TransactionTable.java]: I see what you mean now... I was confused about the "move" part, because there was no `try` block before :)

**Comment by galderz** [comment]
@danberindei Shall we close this PR until you've found a better solution?

**Comment by slaskawi** [comment]
One last call @danberindei :)

**Comment by danberindei** [comment]
I'm giving it one more try today @slaskawi

**Comment by galderz** [comment]
And that day was gone... a while back. Closing for now. Reopen when this is ready.
